src/main/java/Control/Scenario/UC_NewProject.java

```diff
@@ -0,0 +1,173 @@
+                throw new AlreadyExistsException("Project with this  name already exists.");
+            }
+
+            // remove elements that are "" or null
+            removeEmptyElements(projectCreation);
+
+            // check whether all emails provided are already user's
+            if (isProjectNameAndEmailsValid(projectCreation) == false) {
+                jsonResponse.setMessage("Some field's are incorrect or some required are missing.");
+                throw new ProjectCreationException("Some field's are incorrect or some required are missing.");
+            }
+
+            // Get all user's that need addition to Project_User table -- check if they exist
+            List<T_User> listOfUsers = getAllUsersThatWillBeOwners(projectCreation);
+            if (listOfUsers == null) {
+                jsonResponse.setMessage("All e-mails have to be already registered in the project in advance.");
+                throw new ProjectCreationException("All e-mails have to be already registered in the project in advance.");
+            }
+
+            // Create a new project and project_user connections
+            insertProjectAndProjectUsersIntoDatabase(projectCreation, listOfUsers);
+
+            // Success
+            db.afterOkSqlExecution();
+
+            jsonResponse.setStatus(HttpServletResponse.SC_CREATED);
+            jsonResponse.setMessage("User created.");
+        } catch (ProjectCreationException e) {
+            db.afterExceptionInSqlExecution(e);
+
+            jsonResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        } catch (SQLException e) {
+            db.afterExceptionInSqlExecution(e);
+
+            jsonResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            jsonResponse.setMessage("Internal server error.");
+        } catch (AlreadyExistsException e) {
+            db.afterExceptionInSqlExecution(e);
+
+            jsonResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            jsonResponse.setMessage("Project with this name already exists.");
+        }
+
+        return jsonResponse;
+    }
+
+    // Privates
+    private void insertProjectAndProjectUsersIntoDatabase(@NotNull final ProjectCreation projectCreation, @NotNull final List<T_User> listOfUsers) throws SQLException{
+        // Insert new Project
+        I_Project.insert(db.getConn(), db.getPs(), T_Project.CreateFromScratch(projectCreation.getProject_name()));
+
+        int latestId = I_Project.retrieveLatestPerConnectionInsertedID(db.getConn(), db.getPs(), db.getRs());
+
+        // Insert new Connections between project and user
+        for (T_User user:
+             listOfUsers) {
+            Dictionary dict = new Hashtable();
+
+            dict.put(T_Project_user.DBNAME_PROJECTID, latestId);
+            dict.put(T_Project_user.DBNAME_USERID, user.getA_pk());
+
+            I_ProjectUser.insert(db.getConn(), db.getPs(), T_Project_user.CreateFromScratch(dict));
```

Comment:
je nejaka konvencia preco je v T_Project_user user s malym u?

Reply:
No... Neviem. Ono to je také, že to je taká tabuľka spojová a je to také divné hej uznávam